wildfire/goes/utilities.py

```diff
@@ -12,6 +14,68 @@
 }
 REGION_TIME_RESOLUTION_MINUTES = {"M1": 1, "M2": 1, "C": 5, "F": 15}
 
+_logger = logging.getLogger(__name__)
+
+
+def pool_function(function, function_args, num_workers=None):
+    """Run `function` across multiple workers in parallel.
+
+    Parameters
+    ----------
+    function : function
+        Function to pool across multiple threads.
+    function_args : iterable
+        Arguments to iteratively pass to `function` across multiple threads. All elements
+        must be pickleable.
```

Comment:
I think this needs to be clearer for the user or potentially might be too abstract.

For instance:
- You can't pass a function that already calls this function through.
- You would not want to call it on a function that would not benefit from being parralellized (for instance a deterministic search without chunking)

I can list more comments if helpful.